code/modules/mob/living/carbon/give.dm

```diff
@@ -29,32 +22,27 @@
 			if("Yes")
 				if(!I)
 					return
+				if(target.incapacitated() || usr.incapacitated())
+					return
 				if(!Adjacent(target))
 					to_chat(usr, "<span class='warning'> You need to stay in reaching distance while giving an object.</span>")
 					to_chat(target, "<span class='warning'> [usr.name] moved too far away.</span>")
 					return
-				if((usr.hand && usr.l_hand != I) || (!usr.hand && usr.r_hand != I))
+				if((I.flags & NODROP) || (I.flags & ABSTRACT))
+					to_chat(usr, "<span class='warning'>\the [I.name] stays stuck to your hand when you try to give it!</span>")
+					to_chat(target, "<span class='warning'>\the [I.name] stays stuck to [usr.name]'s hand when you try to take it!</span>")
+					return
+				if(I != get_active_hand())
 					to_chat(usr, "<span class='warning'> You need to keep the item in your active hand.</span>")
 					to_chat(target, "<span class='warning'> [usr.name] seem to have given up on giving \the [I.name] to you.</span>")
 					return
 				if(target.r_hand != null && target.l_hand != null)
 					to_chat(target, "<span class='warning'> Your hands are full.</span>")
 					to_chat(usr, "<span class='warning'> Their hands are full.</span>")
 					return
-				else
-					usr.drop_item()
-					if(target.r_hand == null)
-						target.r_hand = I
-					else
-						target.l_hand = I
-				I.loc = target
-				I.layer = 20
-				I.plane = HUD_PLANE
+				usr.unEquip(I)
+				target.put_in_hands(I)
 				I.add_fingerprint(target)
-				src.update_inv_l_hand()
-				src.update_inv_r_hand()
-				target.update_inv_l_hand()
-				target.update_inv_r_hand()
```

Comment:
Oh yes, the days when things like `put_in_hands` and `unEquip` didn't call inventory updates and you had to do it for everything.

Old code still horrifies me to this very day.